src/tagmanager/tm_workspace.c

```diff
@@ -577,6 +521,59 @@ gboolean tm_workspace_create_global_tags(const char *pre_process, const char **i
 	return ret;
 }
 
+/* Creates a list of global tags. Ideally, this should be created once during
+ installations so that all users can use the same file. This is because a full
+ scale global tag list can occupy several megabytes of disk space.
+ @param pre_process_cmd The pre-processing command. This is executed via system(),
+ so you can pass stuff like 'gcc -E -dD -P `gnome-config --cflags gnome`'.
+ @param sources Source files to process. Wildcards such as '/usr/include/a*.h'
+ are allowed.
+ @param tags_file The file where the tags will be stored.
+ @param lang The language to use for the tags file.
+ @return TRUE on success, FALSE on failure.
+*/
+gboolean tm_workspace_create_global_tags(const char *pre_process_cmd, const char **sources,
+	int sources_count, const char *tags_file, TMParserType lang)
+{
+	gboolean ret = FALSE;
+	GList *source_files = lookup_sources(sources, sources_count);
+
+	if (pre_process_cmd)
+		ret = create_global_tags_preprocessed(pre_process_cmd, source_files, tags_file, lang);
+	else
+	{
+		GList *node;
```

Comment:
Maybe stuff the else case also in a static function like the preprocess case?

Reply:
Yeah, that's probably better. Will do.